src/runtime/vdom/vdom-annotations.ts

```diff
@@ -35,8 +35,8 @@ export const insertVdomAnnotations = (doc: Document, staticComponents: string[])
       if (orgLocationNode != null) {
         const nodeRef = orgLocationNode['s-nr'];
 
-        let hostId = nodeRef['s-host-id'];
-        let nodeId = nodeRef['s-node-id'];
+        let hostId = nodeRef?.['s-host-id'];
+        let nodeId = nodeRef?.['s-node-id'];
         let childId = `${hostId}.${nodeId}`;
```

Comment:
With these changes, it's possible for [`childId`](https://github.com/ionic-team/stencil/pull/5231/files#diff-330540c04820aa83be13838701b0dab6068df5ba4a560f7d24382e247f7816edR40) to be assigned to the string `"undefined.undefined"` (as opposed to the runtime failing on attempting to access a property on an `undefined` object (`nodeRef`)). 

We could introduce some funky cases where [`orgLocationNodeId` gets assigned](https://github.com/ionic-team/stencil/pull/5231/files#diff-330540c04820aa83be13838701b0dab6068df5ba4a560f7d24382e247f7816edR67) to either:
- `undefined.undefined.`
- `undefined.undefined.c`

This _probably_ isn't happening today, otherwise I'd expect we'd be seeing a lot more issues in the GH repo.  

Do we know if it's possible for `orgLocationNode.parentElement` to be truthy, but `orgLocationNode['s-nr']` to be falsy? If not, we can probably proceed with @christian-bromann's [change proposal](https://github.com/ionic-team/stencil/pull/5231/files#r1443040184)

Reply:
followed @christian-bromann's suggestion!